### ISSUE TITLE
File contains data in an unknown format.

### ISSUE THREAD

**Issue by peak1995** [CorentinJ/Real-Time-Voice-Cloning#214]
![image](https://user-images.githubusercontent.com/24451894/68924046-cf36ad80-07ba-11ea-98e2-6eb5f45ee97e.png)
Dear sir:
    I'm reproducing your code and meeting a problem. The same corpus and code as yours,but perform"python encoder_preprocess.py  [dataroot]",error in figure above.

**Comment by msebi** [comment]
How did you manage to get pytorch to work? I keep running into:

https://pytorch.org/docs/stable/notes/windows.html#import-error

**Comment by NikkLeiz** [comment]
have same problem 
![image](https://user-images.githubusercontent.com/23361821/68972315-b4455700-0815-11ea-8cef-f2152b2c72cf.png)

**Comment by NiloyPurkait** [comment]
> How did you manage to get pytorch to work? I keep running into:
> 
> https://pytorch.org/docs/stable/notes/windows.html#import-error

managed to solve this by installing cuda 10.1 (had cuda 10.0, and VS2017 before), hope this helps

**Comment by NiloyPurkait** [comment]
The 'demo_cli.py' only uses the ''preprocess_wav" function, which is imported from the encoder.inference script and is defined in the encoder.audio script. It seems preprocessing functions for other file formats are not included yet.

TLDR: convert your files to .wav format to resolve

**Comment by peak1995** [comment]
![image](https://user-images.githubusercontent.com/24451894/69020142-823b1d00-09ee-11ea-8164-7a212c4be501.png)
when I train the encoder_train.py,the process has this error.How can I solve it?

**Comment by golecha** [comment]
Facing the same issue will performing"python encoder_preprocess.py [dataroot]" as the data in VoxCeleb2 is in .m4a files
Also tried running this command "sndfile-play video.m4a" Give the same error : File contains data in an unknown format.

**Comment by Titaniumtown** [comment]
I know the solution is to convert the files to a .wav file, but it takes up way more space; is there a way to remove that requirement of a .wav file?

**Comment by Port0r** [comment]
Even in audio.py it says:
> **param fpath_or_wav:** either a filepath to an audio file (_many extensions are supported, not just .wav_), either the waveform as a numpy array of floats.

And yet...
Well, with ffmpeg and pydub one might alter the script, so it converts any mp3 to wav on the fly. It's a dirty work-around and not time efficient, though...

**Comment by Artem-B** [comment]
The problem is that this fallback path in librosa does not work due to the fact that path is `pathlib.WindowsPath`:  https://github.com/librosa/librosa/blob/master/librosa/core/audio.py#L145

Converting path to string allows file opening with audioread and it will use ffmpeg to load the audio.

**Comment by NealWalters** [comment]
Artem-B - Thanks for your post.  I fixed line 145 as shown below, now on to the next error. 
Fix: 
        if isinstance(str(path), six.string_types):
            warnings.warn('PySoundFile failed. Trying audioread instead.')

And now it's giving the warning on the next line: 
 warnings.warn('PySoundFile failed. Trying audioread instead.')
Caught exception: TypeError("argument of type 'WindowsPath' is not iterable")

**Comment by mhilmiasyrofi** [comment]
Your problem is related to the librosa bug in the latest version. You can refer to this [link](https://github.com/librosa/librosa/issues/1037)

I solve it by converting the path into String
Python 3.6+: `path = os.fspath(path)`
Python 3.4+: `path = str(path)`

**Comment by saraBadawi** [comment]
I've run the script and when I try to upload a file .mp3 
I got Caught exception: RuntimeError("Error opening '.': File contains data in an unknown format.",)
and when I try to upload .wav file I got 
Caught exception: RuntimeError("Error opening '/home/ubuntu/hello-this-my1576942509.wav': System error.",)
Is there something I need to do  I put the voices in /home/ubuntu 
What should I do? and I am using only the terminal

**Comment by SquallAlex** [comment]
I used the gui in windows and it worked with wav files. mp3 is supported but I got the same error. Not sure what to do. Maybe while having a library already working, make a folder name it whatever inside and place some wav and mp3 m4a files try the gui ?

**Comment by saraBadawi** [comment]
I have solved with adding --no_sound arguments to my terminsal `sudo python3 demo_cli.py --no_sound`
That's give me the  demo_output_00.wav.

**Comment by blue-fish** [comment]
Several problems are mentioned in this issue:
* `RuntimeError: Error opening 'filename.mp3': File contains data in an unknown format.` (resolved by #414 to instruct users to install ffmpeg to avoid underlying `Audioread: NoBackendError`)
* Need to convert Path objects to str (resolved by #371)
* `Caught exception: PortAudioError('Error querying device -1',)` when audio playback is attempted in demo_cli.py (resolved by #417)

They have all been resolved in the latest code, so we can finally close this out. Thanks to all here who reported an issue or shared workarounds.

**Comment by skr3178** [comment]
`sudo apt-get install libportaudio2`
helped solve my issue
